AtData/AtPattern/AtPatternY.cxx

```diff
@@ -0,0 +1,146 @@
+#include "AtPatternY.h"
+
+#include <FairLogger.h>
+
+#include <Math/Vector3D.h> // for DisplacementVector3D, operator*
+#include <TEveLine.h>
+#include <TMath.h>
+
+#include <algorithm> // for min
+#include <cmath>     // for cos, sin, pow, sqrt, acos, atan, fabs
+
+using namespace AtPatterns;
+
+ClassImp(AtPatternY);
+
+AtPatternY::AtPatternY() : AtPattern(4) {}
+
+TEveLine *AtPatternY::GetEveLine() const
+{
+   return nullptr;
+}
+
+XYZVector AtPatternY::GetDirection(int line) const
+{
+   if (line < 3) {
+      return {fPatternPar[3 + 3 * line], fPatternPar[4 + 3 * line], fPatternPar[5 + 3 * line]};
+   } else {
+      LOG(error) << "Line number, " << line << ", is greater than the number of actual lines, 3.";
+      return {0, 0, 0};
+   }
+}
+
+XYZPoint AtPatternY::ClosestPointOnPattern(const XYZPoint &point) const
+{
+   auto p = GetVertex();
+   auto d = GetDirection(0);
+
+   return p;
+}
+
+Double_t AtPatternY::DistanceToPattern(const XYZPoint &point) const
+{
+   auto vec = GetVertex() - point;
+   double minDist = 1000;
+   for (int i = 0; i < 3; i++) {
+      double dist2;
+      if ((i == 0 && point.Z() > GetVertex().Z()) || (i > 0 && point.Z() < GetVertex().Z())) {
+         auto nD = GetDirection(i).Cross(vec);
+         dist2 = nD.Mag2() / GetDirection(i).Mag2();
+      } else {
+         dist2 = vec.Mag2();
+      }
+      if (dist2 < minDist) {
+         minDist = dist2;
+      }
+   }
+
+   return std::sqrt(minDist);
+}
+
+void AtPatternY::DefinePattern(const std::vector<XYZPoint> &points)
+{
+   // std::cout << "making pattern" << std::endl;
+   if (points.size() != fNumPoints)
+      LOG(error) << "Trying to create model with wrong number of points " << points.size();
+
+   int zPosIndx[4] = {};
+   double zPos[4] = {};
+
+   for (int i = 0; i < 4; i++) {
+      zPosIndx[i] = i;
+      zPos[i] = points[i].Z();
+      // std::cout << "1 " << i << std::endl;
+   }
+   for (int i = 0; i < 3; i++) {
+      // std::cout << "2 " << i << std::endl;
+      for (int r = 0; r < 3; r++) {
+         // std::cout << "2 " << i << std::endl;
+         // std::cout << 4 - i << std::endl;
+         // std::cout << "3 " << r << std::endl;
+         if (zPos[r] > zPos[r + 1]) {
+            std::swap(zPos[r], zPos[r + 1]);
+            std::swap(zPosIndx[r], zPosIndx[r + 1]);
+         }
+      }
+   }
+   for (int i = 0; i < 4; i++) {
```

Comment:
We should remove these unused loops (probably the compiler does already but 🤷🏻 )